### PR TITLE
Tests for logs routes.

### PR DESCRIPTION
**Pull request by jander-msft**
This change adds tests for the `GET /logs` and `POST /logs` routes and tests each aspect of those routes. It tests these routes using newline-delimited JSON responses since those are relatively easy to process.

### CODE REVIEW

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+            }
+        }
+
+        private static LogEntry CreateTraceEntry(string category)
+        {
+            return new LogEntry()
+            {
+                Category = category,
+                EventId = 1,
+                EventName = string.Empty,
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs:475]: All of the log events being received by the tests show an empty event name for some reason. I will investigate this and try to correct it in this PR.

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+                // The other lines contain callstack information, which will vary between machines.
+                string firstLine = actual.Exception.Split(Environment.NewLine)[0];
+                Assert.Equal(expected.Exception, firstLine);
+            }
+
+            Assert.Equal(expected.LogLevel, actual.LogLevel);
+
+            Assert.Equal(expected.Message, actual.Message);
+
+            // TODO: Work on scope verification
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs:444]: I'd like to add scope information to the log events of the target application, however I'll do that in a subsequent change.

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/HttpApi/ApiClient.cs [hunk 3]
```diff
@@ -192,6 +195,107 @@ private async Task<ResponseStreamHolder> CaptureDumpAsync(string processKey, Dum
...
+        /// POST /logs/{uid}?durationSeconds={duration}
+        /// </summary>
+        public Task<ResponseStreamHolder> CaptureLogsAsync(Guid uid, TimeSpan duration, LogsConfiguration configuration, CancellationToken token)
+        {
+            return CaptureLogsAsync(uid.ToString("D"), duration, configuration, token);
+        }
+
+        private Task<ResponseStreamHolder> CaptureLogsAsync(string processKey, TimeSpan duration, LogsConfiguration configuration, CancellationToken token)
+        {
+            StringBuilder routeBuilder = new();
```
Comment (wiktork) [inline src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/HttpApi/ApiClient.cs:252]: nit: Maybe add a utility function to compose common routes?

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs (hunk 1 above)
Comment (wiktork) [reply src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs:475]: Many events do not have names, but I am surprised none of them do.

**Review by wiktork — APPROVED** [review]

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs (hunk 1 above)
Comment (jander-msft) [reply src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs:475]: All of the log events that are being generated in the logger scenario are explicitly written by the test app and have Event IDs with both ID and name values.

Comment (jander-msft) [reply src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs:523]: Fix for this is https://github.com/dotnet/diagnostics/pull/2291

**Comment by jander-msft** [comment]
Test runs:
1. [1142349](https://dev.azure.com/dnceng/public/_build/results?buildId=1142349&view=results): All tests passed and one passed on rerun. Reran MacOS Release build leg due to a single test unable to start dotnet-monitor (nothing in stdout or stderr for dotnet-monitor).
2. [1142464](https://dev.azure.com/dnceng/public/_build/results?buildId=1142464&view=results): All tests passed.

**Pull request merged**